lms/djangoapps/grades/tests/test_signals.py

```diff
@@ -148,3 +158,91 @@ def test_score_reset_bad_user(self):
+        self.user = UserFactory()
+        self.course = CourseFactory.create(
+            org='edx',
+            name='course',
+            run='run',
+            metadata={'enable_subsection_grades_saved': enable_subsection_grades})
+
+        self.chapter = ItemFactory.create(parent=self.course, category="chapter", display_name="Chapter")
+        self.sequential = ItemFactory.create(parent=self.chapter, category='sequential', display_name="Open Sequential")
+        self.problem = ItemFactory.create(parent=self.sequential, category='problem', display_name='problem')
+
+        self.score_changed_kwargs = {
+            'points_possible': 10,
+            'points_earned': 5,
+            'user_id': self.user.id,
+            'course_id': unicode(self.course.id),  # pylint: disable=no-member
+            'usage_id': unicode(self.problem.location),
+        }
+        # pylint: enable=attribute-defined-outside-init
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_subsection_grade_updated_on_signal(self, default_store):
+        with self.store.default_store(default_store):
+            self.set_up_course_and_user(True)
+            with check_mongo_calls(2) and self.assertNumQueries(18):
+                recalculate_subsection_grade_handler(None, **self.score_changed_kwargs)
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_query_count_does_not_change_with_more_problems(self, default_store):
+        with self.store.default_store(default_store):
+            self.set_up_course_and_user(True)
+            ItemFactory.create(parent=self.sequential, category='problem', display_name='problem2')
+            ItemFactory.create(parent=self.sequential, category='problem', display_name='problem3')
+            with check_mongo_calls(2) and self.assertNumQueries(18):
+                recalculate_subsection_grade_handler(None, **self.score_changed_kwargs)
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_subsection_grades_not_enabled_on_course(self, default_store):
+        with self.store.default_store(default_store):
+            self.set_up_course_and_user(False)
+            with check_mongo_calls(2) and self.assertNumQueries(1):
+                recalculate_subsection_grade_handler(None, **self.score_changed_kwargs)
+
+    @ddt.data(
+        (ModuleStoreEnum.Type.mongo, True),
+        (ModuleStoreEnum.Type.split, True),
+        (ModuleStoreEnum.Type.mongo, False),
+        (ModuleStoreEnum.Type.split, False),
+    )
+    @ddt.unpack
+    def test_score_changed_sent_with_feature_flag(self, default_store, feature_flag):
+        with patch.dict('django.conf.settings.FEATURES', {'ENABLE_SUBSECTION_GRADES_SAVED': feature_flag}):
+            with self.store.default_store(default_store):
+                self.set_up_course_and_user(True)
+                with check_mongo_calls(0) and self.assertNumQueries(19 if feature_flag else 1):
```

Comment:
@nasthagiri @efischer19 this passes locally, but on Jenkins the SQL counts are all off by 1:

https://build.testeng.edx.org/job/edx-platform-test-subset/558167/console

Is there a setting that differs between devstack and test that could cause this? It looks like the difference is that a query on the `lti_provider_gradedassignment` table is being executed once locally and twice on Jenkins.